custom_components/alexa_media/alarm_control_panel.py

```diff
@@ -351,11 +227,10 @@ def supported_features(self) -> int:
         return SUPPORT_ALARM_ARM_AWAY
 
     @property
-    def available(self):
-        """Return the availability of the device."""
-        return self._available
+    def assumed_state(self) -> bool:
+        return self._login.session.closed
```

Comment:
Is this the right test? You may be right but the component can recycle the connection on reconnect.

Reply:
This was the test used before. However, now that you have pointed it out, the DataUpdateCoordinator does make it possible to write a more robust test that avoids needing to even think about login.

I believe that this is now a better check, but it is a slight change in behavior. Previous to my changes, assumed_state() was true only if there was a login failure on the last refresh. Now, assumed_state() will be true if the last refresh failed for any reason.

Reply:
Makes sense. I believe HA uses that now for all the assumed_state calls with an update coordinator.